ouroboros-consensus-test/src/Test/ThreadNet/Network.hs

```diff
@@ -1044,29 +1044,10 @@ runThreadNetwork systemTime ThreadNetworkArgs
       -- Create a 'DiskLedgerView' to be used in 'forkTxProducer'. This function
       -- needs the 'DiskLedgerView' to elaborate a complete UTxO set to generate
       -- transactions.
-      let
-        extractBsValueHandle ::
-             StaticEither
-               'False
-               ( LedgerBackingStoreValueHandle m (ExtLedgerState blk)
-               , LedgerDB' blk
-               , m ()
-               )
-               (Either
-                 (Point blk)
-                 ( LedgerBackingStoreValueHandle m (ExtLedgerState blk)
-                 , LedgerDB' blk
-                 , m ())
-               )
-          -> ( LedgerBackingStoreValueHandle m (ExtLedgerState blk)
-             , LedgerDB (ExtLedgerState blk)
-             , m ()
-             )
-        extractBsValueHandle = \case
-            StaticLeft bsValueHandle -> bsValueHandle
-      seBsValueHandle <-
-          fmap extractBsValueHandle
-        $ ChainDB.getLedgerBackingStoreValueHandle chainDB registry (StaticLeft ())
+      let getValueHandle = do
+            bsvh <- ChainDB.getLedgerBackingStoreValueHandle chainDB registry (StaticLeft ())
+            pure $ mkDiskLedgerView $ case bsvh of
+              StaticLeft v -> v
```

Comment:
What do you think of adding these functions to `Ouroboros.Consensus.Util`:

```haskell
staticFromLeft :: StaticEither 'False l r -> l
staticFromLeft = \case (StaticLeft x) -> x

staticFromRight :: StaticEither 'True l r -> r
staticFromRight = \case (StaticRight x) -> x
```

Reply:
I like the idea.